pallets/xyk/src/lib.rs

```diff
@@ -1654,6 +1648,12 @@ impl<T: Config> XykFunctionsTrait<T::AccountId> for Pallet<T> {
 		second_asset_id: Self::CurrencyId,
 		second_asset_amount: Self::Balance,
 	) -> DispatchResult {
+		ensure!(
+			!T::DisabledTokens::contains(&first_asset_id) &&
+				!T::DisabledTokens::contains(&second_asset_id),
+			Error::<T>::FunctionNotAvailableForThisToken
+		);
+
```

Comment:
we are using the DisabledTokens trait to pass `xykMeta.pool_creation_disabled`
move the check here so that also other pallets using the xyk API have to pass